### DISCUSSION TITLE
Controls events propagation

### DISCUSSION THREAD

**Discussion by xgebi** [storybookjs/storybook#26806] [Help] [answered]
### Summary

In the project I'm working on we have components that close when user clicks elsewhere, it's done through document listener on click. The issue is with storybook controls. Is there a way for controls not to propagate their events?

When I say controls I mean these:

<img width="663" alt="Screenshot of controls addon in Storybook" src="https://github.com/storybookjs/storybook/assets/2587524/9ea1e2d4-0a27-4d0d-ba51-04c3a7985dc5">

At the moment we have a workaround in our components where we detect if the click was from Storybook or not.

**Comment by Marklb** [comment, marked as answer]
Can you provide a more clear explanation of what is happening and what you want to happen?

My assumption is that you are wanting your component to not detect the click events from the Storybook UI. Since you mention controls specifically, I don't know if I am right.

If I am right, I can describe one solution that I use, which may or may not be simple in your framework. When I have a story for a modal, I don't want it to close, but by default the close when clicking anywhere outside it's container. I have options to disable most of those, but as a generic catch-all for stories I use dependency injection to swap the handler of the close method.

I use Angular and I have a class that is injected in my modal that handles various things, like closing the modal. It is simple to replace most things in Angular's injector, so in stories I swap that class with another class that has the `close()` method, but nothing is in the body of the method.

Depending on your situation, there may be a better solution. That is just how I do what I think you are trying to achieve.

If you need the story to instead be isolated in it's own document, there is also the option of disabling inline rendering. There are performance and limitation downsides to that, but there is an isolation benefit to loading each story in it's own iframe. https://storybook.js.org/docs/api/doc-block-story#inline

**Reply by xgebi** [reply]
I have a lit web component that should close when a user clicks anywhere else than inside the component. At the same time when the component is in opened state and user changes something in storybook's controls by clicking on them them, the component should remain open.

I'll try the inline thing and will let you know how that went.

**Reply by xgebi** [reply]
So the inline does work but the controls stopped working, there's a note in docs that it's a known issue
